sandbox.go

```diff
@@ -644,13 +644,6 @@ func (sb *sandbox) SetKey(basePath string) error {
 	sb.Lock()
 	sb.osSbox = osSbox
 	sb.Unlock()
-	defer func() {
-		if err != nil {
-			sb.Lock()
```

Comment:
Shouldnt we instead perform the cleanup here and not rely on external caller to do  the cleanup ?
Its generally a good practice to cleanup any locally allocated resource (GetSandboxForExternalKey) in the defer.